ui/components/app/confirm/info/row/address.tsx

```diff
@@ -1,41 +1,79 @@
-import React from 'react';
-import {
-  AvatarAccount,
-  Box,
-  Text,
-  AvatarAccountSize,
-} from '../../../../component-library';
+import { NameType } from '@metamask/name-controller';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
 import {
   AlignItems,
   BorderColor,
   Display,
   FlexDirection,
   TextColor,
 } from '../../../../../helpers/constants/design-system';
-import { shortenAddress } from '../../../../../helpers/utils/util';
-import Tooltip from '../../../../ui/tooltip/tooltip';
+import { getPetnamesEnabled } from '../../../../../selectors';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  Box,
+  Text,
+} from '../../../../component-library';
+import NicknamePopovers from '../../../modals/nickname-popovers';
+import Name from '../../../name/name';
+import { useFallbackDisplayName } from './hook';
 
 export type ConfirmInfoRowAddressProps = {
   address: string;
+  isSnapUsingThis?: boolean;
 };
 
 export const ConfirmInfoRowAddress = ({
   address,
-}: ConfirmInfoRowAddressProps) => (
-  <Box
-    display={Display.Flex}
-    flexDirection={FlexDirection.Row}
-    alignItems={AlignItems.center}
-  >
-    <AvatarAccount
-      address={address}
-      size={AvatarAccountSize.Xs}
-      borderColor={BorderColor.transparent}
-    />
-    <Tooltip position="top" title={address} interactive>
-      <Text marginLeft={2} color={TextColor.inherit}>
-        {shortenAddress(address)}
-      </Text>
-    </Tooltip>
-  </Box>
-);
+  isSnapUsingThis,
+}: ConfirmInfoRowAddressProps) => {
+  const isPetNamesEnabled = useSelector(getPetnamesEnabled);
+  const { displayName, hexAddress } = useFallbackDisplayName(address);
+  const [isNicknamePopoverShown, setIsNicknamePopoverShown] = useState(false);
+
+  return (
+    <Box
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      alignItems={AlignItems.center}
+    >
+      {
+        // PetNames on this component are disabled for snaps until the `<Name />`
+        // component can support variations. See this comment for context: //
+        // https://github.com/MetaMask/metamask-extension/pull/23487#discussion_r1525055546
+        isPetNamesEnabled && !isSnapUsingThis ? (
+          <Name value={hexAddress} type={NameType.ETHEREUM_ADDRESS} />
+        ) : (
+          <>
+            <Box
+              display={Display.Flex}
+              flexDirection={FlexDirection.Row}
+              alignItems={AlignItems.center}
+              onClick={() => setIsNicknamePopoverShown(true)}
```

Comment:
Please don't use arrow functions directly in onClick handlers. They should be defined alongside hooks for performance reasons.